src/tests/ftest/util/nvme_utils.py

```diff
@@ -0,0 +1,371 @@
+
+                # Wait for rebuild to start
+                self.pool.wait_for_rebuild(True)
+
+                # Wait for rebuild to complete
+                self.pool.wait_for_rebuild(False)
+
+    def start_ior_load(self):
+        """
+        Method to Fill up the server. It will get the maximum Storage space and
+        create the pool.Fill up the server based on % amount given using IOR.
+
+        Returns:
+            None
+        """
+        #Method to get the storage capacity.Replace with dmg options in future
+        #when it's available. This is time consuming so store the size in
+        #file to avoid rerunning the same logic for future test cases.
+        #This will be only run for first test case.
+
+        avocao_tmp_dir = os.environ['AVOCADO_TESTS_COMMON_TMPDIR']
+        capacity_file = os.path.join(avocao_tmp_dir, 'storage_capacity')
+        if not os.path.exists(capacity_file):
+            #Stop server but do not reset.
+            self.stop_servers_noreset()
+            total_nvme_capacity = self.get_server_capacity()
+            with open(capacity_file,
+                      'w') as _file: _file.write('{}'
+                                                 .format(total_nvme_capacity))
+            #Start the server.
+            self.start_servers()
+        else:
+            total_nvme_capacity = float(open(capacity_file)
+                                        .readline().rstrip())
+
+        print("Server Stoarge capacity = {}GB".format(total_nvme_capacity))
+        # Create a pool
+        self.pool = TestPool(self.context, dmg_command=self.get_dmg_command())
+        self.pool.get_params(self)
+        self.pool.nvme_size.update('{}G'.format(total_nvme_capacity))
+        self.pool.create()
+        print("Pool Usage Percentage - Before - {}"
+              .format(self.pool.pool_percentage_used()))
+
+        # Create the IOR threads
+        job = threading.Thread(target=self.start_ior_thread,
+                               kwargs={"results":self.out_queue,
+                                       "operation": 'Write'})
+        # Launch the IOR thread
+        job.start()
+
+        #Set NVMe device faulty if it's set
+        if self.set_faulty_device:
+            time.sleep(20)
+            #Set the device faulty
+            self.set_device_faulty()
+
+        # Wait to finish the thread
+        job.join()
+
+        # Verify the queue and make sure no FAIL for any IOR run
+        while not self.out_queue.empty():
+            if self.out_queue.get() == "FAIL":
+                self.fail("FAIL")
+
+        print("pool_percentage_used -- After -- {}"
+              .format(self.pool.pool_percentage_used()))
+
+        #Check nvme-health command works
+        try:
+            self.dmg.hostlist = self.hostlist_servers
+            self.dmg.storage_query_nvme_health()
+        except CommandFailure as error:
+            self.fail("dmg nvme-health failed {}".format(details))
```

Comment:
(pylint-missing-final-newline) Final newline missing
(pylint-mixed-line-endings) Mixed line endings LF and CRLF
(pylint-undefined-variable) Undefined variable 'details'